packages/next/src/builders/build/lib/create-package-json.ts

```diff
@@ -2,23 +2,63 @@ import { BuilderContext } from '@angular-devkit/architect';
 import { writeFileSync } from 'fs';
 import { join } from 'path';
 import { readJsonFile } from '@nrwl/workspace';
+import { createProjectGraph } from '@nrwl/workspace/src/core/project-graph';
+import { calculateProjectDependencies } from '@nrwl/workspace/src/utils/buildable-libs-utils';
 import { NextBuildBuilderOptions } from '../../../utils/types';
 
+function getProjectDeps(context: BuilderContext, rootPackageJson: any) {
+  const projGraph = createProjectGraph();
+  const { dependencies: deps } = calculateProjectDependencies(
+    projGraph,
+    context
+  );
+  const depNames = deps
+    .map((d) => d.node)
+    .filter((node) => node.type === 'npm')
+    .map((node) => node.name.replace('npm:', ''));
```

Comment:
You can read from `node.data.packageName` which is safer in case we do change the naming convention later.